pkg/bootstrap/platform/gcp.go

```diff
@@ -251,15 +252,19 @@ func waitForMetadataSuppliers(suppliers []metadataSupplier, md map[string]string
 
 // Converts a GCP zone into a region.
 func zoneToRegion(z string) (string, error) {
-	// Zones are in the form <region>-<zone_suffix>, so capture everything but the suffix.
-	re := regexp.MustCompile("(.*)-.*")
-	m := re.FindStringSubmatch(z)
+	m := zoneRE.FindStringSubmatch(z)
 	if len(m) != 2 {
 		return "", fmt.Errorf("unable to extract region from GCP zone: %s", z)
 	}
 	return m[1], nil
 }
 
+var (
+	// Zones are in the form <country>-<region>-<zone_suffix>, so capture everything but the suffix.
```

Comment:
How did we change from "Zones are in the form <region>-<zone_suffix>" to "<country>-<region>-<zone_suffix>"

Reply:
I think it was from some misassumption, which was not considering the regional cluster.
In regional GKE cluster, the location will be given like "us-central1". In that case, it is parsed as 
```
region: "us"
zone: "us-central1"
```

Reply:
`us-central1` is not `<country>-<region>-<zone_suffix>`? that pattern has 2 `-` in it?

Reply:
Oh. I mean the previous impl (before this PR.)

BTW, I realized that I misunderstand your question. 

So, I applied changes so that it is falling back to "us-central1" -> "us-central1-unknown"